src/mono/wasm/wasm.proj

```diff
@@ -259,6 +259,7 @@
       <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Debug'"  >$(CMakeConfigurationEmccFlags)</CMakeConfigurationLinkFlags>
       <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Release'">-O2</CMakeConfigurationLinkFlags>
       <CMakeConfigurationLinkFlags>$(CMakeConfigurationLinkFlags) -s EXPORT_ES6=1</CMakeConfigurationLinkFlags>
+      <CMakeConfigurationLinkFlags>$(CMakeConfigurationLinkFlags) -Wl,--allow-undefined</CMakeConfigurationLinkFlags>
```

Comment:
could we rather deal with `icudt68_dat` explicitly here and drop `--allow-undefined` ?

Reply:
I didn't found how, because (as described in the summary)

- The list of allowed undefined symbols is controlled by the emscripten [emcc.py](https://github.com/emscripten-core/emscripten/blob/73ab1c59529692d838125f4b6df87c62d8848fe5/emcc.py#L2994-L2996) -> [building.py](https://github.com/emscripten-core/emscripten/blob/73ab1c59529692d838125f4b6df87c62d8848fe5/tools/building.py#L154) and comes in computed list from library functions in [dotnet.es6.lib.js](https://github.com/dotnet/runtime/blob/116636d48b3fab46bb8d0fee5fe2405f5397929b/src/mono/wasm/runtime/es6/dotnet.es6.lib.js#L128).

Any ideas?

Reply:
could we have dummy js function in `dotnet.es6.lib.js` ?

Reply:
The `--allow-undefined-file=` is already placed by emcc. It is described in the summary and in my first reply in this thread.

Reply:
I pushed a change make add the missing libicudata.a to the cmakelists.txt which I think will fix it

Reply:
I should be able to rid of it in the wasm.proj by linking the lib in cmake and ensuring that it is not reachable.
For user symbol I would to get of undefined symbols much earlier than in JS.